File: lib/models/minimal_game.dart
import 'game.dart';

class MinimalGame {
  int id;
  String name;
  String date;
  String cover;
  String platforms;

  MinimalGame({
    this.id = 0,
    this.name = "",
    this.date = "",
    this.cover = "",
    this.platforms = "",
  });

  MinimalGame.fromJson(Map<dynamic, dynamic> json)
      : id = json['id'],
        name = json['name'],
        date = json['date'],
        cover = json['cover'],
        platforms = json['platforms'];

  MinimalGame.fromGame(Game game)
      : id = game.id,
        name = game.name,
        date = game.date,
        cover = game.cover,
        platforms = game.platforms;
}

File: functions/src/models/minimal_game.ts
class MinimalGame {
  id: number;
  name: string;
  date: string;
  cover: string;
  platforms: string;

  constructor(
    id = 0,
    name = "",
    date = "",
    cover = "",
    platforms = "",
  ) {
    this.id = id;
    this.name = name;
    this.date = date;
    this.cover = cover;
    this.platforms = platforms;
  }
}

export = MinimalGame;
